examples/tutorial_server_eventfilter.c

```diff
@@ -0,0 +1,282 @@
+                                                 &Estatus, &UA_TYPES[UA_TYPES_DATETIME]);
+            serverId = UA_STRING("SampleServer");
+            UA_Server_writeObjectProperty_scalar(server, *outId, UA_QUALIFIEDNAME(0, "ServerId"),
+                                                 &serverId, &UA_TYPES[UA_TYPES_DATETIME]);
+            clientUserId = UA_STRING("SampleClient");
+            UA_Server_writeObjectProperty_scalar(server, *outId, UA_QUALIFIEDNAME(0, "ClientUserId"),
+                                                 &clientUserId, &UA_TYPES[UA_TYPES_DATETIME]);
+            clientAuditEntryId = UA_STRING("SampleAudit");
+            UA_Server_writeObjectProperty_scalar(server, *outId, UA_QUALIFIEDNAME(0, "ClientAuditEntryId"),
+                                                 &clientAuditEntryId, &UA_TYPES[UA_TYPES_DATETIME]);
+            statusCode = UA_STATUSCODE_GOOD;
+            UA_Server_writeObjectProperty_scalar(server, *outId, UA_QUALIFIEDNAME(0, "StatusCodeId"),
+                                                 &statusCode, &UA_TYPES[UA_TYPES_LOCALIZEDTEXT]);
+            break;
+        default:
+            eventSourceName = UA_STRING("Server");
+            break;
+    }
+    UA_Server_writeObjectProperty_scalar(server, *outId, UA_QUALIFIEDNAME(0, "SourceName"),
+                                         &eventSourceName, &UA_TYPES[UA_TYPES_STRING]);
+    return UA_STATUSCODE_GOOD;
+}
+
+/**
+ * Triggering an event
+ * ^^^^^^^^^^^^^^^^^^^
+ * First a node representing an event is generated using ``setUpEvent``. Once our event is good to go, we specify
+ * a node which emits the event - in this case the server node. We can use ``UA_Server_triggerEvent`` to trigger our
+ * event onto said node. Passing ``NULL`` as the second-last argument means we will not receive the `EventId`.
+ * The last boolean argument states whether the node should be deleted. */
+static UA_StatusCode
+generateEventsMethodCallback(UA_Server *server,
+                         const UA_NodeId *sessionId, void *sessionHandle,
+                         const UA_NodeId *methodId, void *methodContext,
+                         const UA_NodeId *objectId, void *objectContext,
+                         size_t inputSize, const UA_Variant *input,
+                         size_t outputSize, UA_Variant *output) {
+    UA_LOG_INFO(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND, "Creating event");
+    UA_StatusCode retval;
+    UA_NodeId* eventNodeId;
+    eventNodeId = (UA_NodeId*)
+        UA_Array_new(eventTypesCount, &UA_TYPES[UA_TYPES_NODEID]);
+    /* set up event */
+    for(size_t i = 0 ; i < eventTypesCount; i++) {
+        retval = setUpEvent(server, &eventNodeId[i], eventTypes[i], false,i);
+        if(retval != UA_STATUSCODE_GOOD) {
+            UA_LOG_WARNING(UA_Log_Stdout, UA_LOGCATEGORY_USERLAND,
+                           "Creating event failed. StatusCode %s", UA_StatusCode_name(retval));
+            return retval;
```

Comment:
If you return here i think we will lose the allocated memory for "eventNodeId". This is a general hint: Check on each error return if there is some cleanup needed.